feat: pass `use_fast` parameter to `get_tokenizer`

nikitajz opened the pull request:
A proposed solution for the issue [#105](https://github.com/Tiiiger/bert_score/issues/105)

felixgwu commented on bert_score/utils.py:
```diff
@@ -272,12 +272,12 @@ def get_model(model_type, num_layers, all_layers=None):
     if LooseVersion(trans_version) >= LooseVersion("4.0.0"):
-        tokenizer = AutoTokenizer.from_pretrained(model_type, use_fast=False)
+        tokenizer = AutoTokenizer.from_pretrained(model_type, use_fast=use_fast)
     else:
```
Maybe add `assert not use_fast` here because it's not available?

nikitajz replied:
added. But beware, `assert` can be disabled during optimization hence its usage is arguable

felixgwu commented on tests/test_score_function.py:
```diff
@@ -92,7 +92,7 @@ def test_score_rescale(self):
             (F - torch.tensor([0.901383399963379, 0.871010780334473, 0.503565192222595])).abs_().max() < EPS
         )
 
-    def test_idf_score_rescale(self):
```
Can you keep the old `test_idf_score_rescale` so that we can detect if only the rescaling is not working?

felixgwu reviewed (changes requested):
Looks great! There are only a few tiny changes that need to be made.

nikitajz commented:
@felixgwu after a few attempts I'm still getting failed tests due to OOM (`OSError: [Errno 12] Cannot allocate memory`) even for old tests without a fast tokenizer. I've seen the related issue but lost the link that Travis CI decreased memory limit to 3.5 Gb (from 7Gb) which might be related. Though it works fine on my laptop (except for one unrelated test due to commented model 'scibert-scivocab-uncased' which I mentioned earlier).
Apart from that, seems [get_idf_dict](https://github.com/Tiiiger/bert_score/blob/master/bert_score/utils.py#L315-L335) worth refactoring (at least for fast tokenizers) to use `tokenizer.encode_batch` instead of paralleling `tokenizer.encode`, see related [issue](https://github.com/huggingface/tokenizers/issues/398)

felixgwu reviewed (approved):
Everything looks good to me now. I can either merge it or wait if you would like to spend more time on fixing the Travis CI. I really appreciate your help!

nikitajz commented:
ok, let see if I can devote more time to fixing the tests, otherwise, feel free to merge it.
Also, I'd recommend changing the model in the tests from `roberta-large` to something lighter, e.g. `distilroberta-base`

nikitajz commented:
I've refactored the tests a bit. I also added the missing assert statement to `test_multi_refs_working`, please check if it is correct. I temporarily added skip to test `test_score_en_sci` which fails due to commented model in utils. All tests pass successfully locally.
Unfortunately, I didn't fix failing tests on Travis CI quickly. As far as I can tell, there are two possible options:
- migrate to another CI hosting with more powerful VMs in the free tier
- replace model `roberta-large` in the test with lightweight, something like `distilroberta-base`, but this requires also changing some tests behaviour, e.g. `test_multi_refs` where a model is not specified explicitly, but chosen by language specified. Hence I didn't proceed with this option.

felixgwu commented:
It looked good to me, so I just merged them. I would like to keep using `roberta-large` in the tests for now since it is the default model. Thanks again for your help!
